llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -276,6 +276,7 @@ static bool isZeroingInactiveLanes(SDValue Op) {
     if (ISD::isConstantSplatVectorAllOnes(Op.getNode()))
       return true;
     return false;
+  case ISD::SPLAT_VECTOR:
```

Comment:
Is the code above (`isConstantSplatVectorAllOnes`) needed anymore. Or at all?

Before this patch it could handle
```
  %.splatinsert = insertelement <vscale x 4 x i1> poison, i1 1, i64 0
  %.splat = shufflevector <vscale x 4 x i1> %.splatinsert, <vscale x 4 x i1> poison, <vscale x 4 x i32> zeroinitializer
  %0 = tail call <vscale x 16 x i1> @llvm.aarch64.sve.convert.to.svbool.nxv4i1(<vscale x 4 x i1> %.splat)
```

but not

```
  %.splatinsert = insertelement <vscale x 4 x i1> poison, i1 0, i64 0
  %.splat = shufflevector <vscale x 4 x i1> %.splatinsert, <vscale x 4 x i1> poison, <vscale x 4 x i32> zeroinitializer
  %0 = tail call <vscale x 16 x i1> @llvm.aarch64.sve.convert.to.svbool.nxv4i1(<vscale x 4 x i1> %.splat)
```

which makes me a bit skeptical about it in general. Now it becomes meaningless, except when you have a sequence of bitcasts ending with a `splat_vector`, but then maybe you should handle such a sequence in this patch?

Reply:
Yes. I didn't quite understood the purpose of that line as well, but just in case I missed smth I left it there. Maybe it was written before splat_vector was implemented using `WHILELO` instruction, so you couldn't just assume the other lanes would be zeroed. I will remove it and rewrite my current fix to handle bitcast chain before splat_vector.